src/netaddress.cpp

```diff
@@ -29,7 +32,18 @@ CNetAddr::BIP155Network CNetAddr::GetBIP155Network() const
     case NET_IPV6:
         return BIP155Network::IPV6;
     case NET_ONION:
-        return BIP155Network::TORV2;
+        switch (m_addr.size()) {
+        case ADDR_TORV2_SIZE:
+            return BIP155Network::TORV2;
+        case ADDR_TORV3_SIZE:
+            return BIP155Network::TORV3;
+        default:
+            assert(false);
```

Comment:
The length cannot be something else than `ADDR_TORV2_SIZE` or `ADDR_TORV3_SIZE`. There is no code path that would lead to such scenario and if it happens it means some memory corruption or some future bug. It is the same as `m_net` being equal to `NET_MAX` for example.

Reply:
> // length should have been checked before calling this function

I disagree; length is part of the `CNetAddr` objects, so this precondition is just restating an implicit "can only be called on a valid `CNetAddr` object".